### DISCUSSION TITLE
act on press instead of release

### DISCUSSION THREAD

**Discussion by bkbartje** [arendst/Tasmota#10926] [Q&A] [answered]
Hello,
I have this sonoff wall switch
https://www.sonoff.nl/a-56200593/wifi-rf-touchschakelaars/sonoff-wifi-enkelpolige-schakelaar-zonder-nul-wit/#description
flashed with tasmota,
everything works as expected the I found out the button reacts on button release instead of button press.
This somehow gives an unnatural response.
is it possible to have this button react on press?
I checked SWITCHMODE, but I'm not sure to which mode I should change it.
or is it just a shortcomming off this button that there is no event triggered on press.

**Comment by ascillato** [comment, marked as answer]
> is it possible to have this button react on press?

Hi,

Yes it is possible and already available in Tasmota by a simple configuration in the Tasmota Console.
Please check the docs at https://tasmota.github.io/docs/Commands/#setoptions
Search for the commands `SetOption1` and `SetOption13` to achieve the behaviour that fits better for your use-case.

Besides that, you can find the full explanation for Switches and Buttons usage in Tasmota on the docs at https://tasmota.github.io/docs/Buttons-and-Switches/#changing-default-functionality in order to better fit any requirement you need for your project.

**Reply by bkbartje** [reply]
Thank you SetOption13 is exactly what I need.

and those console commands are really easy to use from the web interface.
